### PR TITLE
[Improve]Improve streampark-common module base on [6 Testing]

### LINKED ISSUE #3063: [Improve] Improve `streampark-common` module base on [6 Testing]
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Improve `streampark-common` module base on [6 Testing](https://streampark.apache.org/community/submit_guide/code_style_and_quality_guide#6-testing)

### Usage Scenario

_No response_

### Related issues

#2986 

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhengkezhou1**
## What changes were proposed in this pull request

Issue Number: close #3063 

## Brief change log

*add scalatest dependency to streampark-common pom.xml*
*add mockito-scala_2.12 dependency to streampark-common pom.xml*

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes)

### CODE REVIEW

**Comment by zzzk1** [comment]
QQ截图20240128191324
*There are a lot of tool classes here but I find some of them are not being used, is it necessary to write test cases for all of them?*

**Comment by caicancai** [comment]
I will start reviewing this PR if no one reviews it

File: streampark-common/src/test/scala/org/apache/streampark/common/fs/HdfsOperatorTest.scala [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+import org.apache.commons.codec.digest.DigestUtils
+import org.apache.commons.io.{FileUtils, IOUtils}
+import org.junit.jupiter.api.Assertions.{assertDoesNotThrow, assertEquals, assertFalse, assertNotEquals, assertThrows, assertTrue}
+import org.junit.jupiter.api.Test
+
+import java.io.{File, FileInputStream}
+import java.nio.file.{Files, Paths}
+
+object HdfsOperatorTest {
```
Comment (caicancai) [inline streampark-common/src/test/scala/org/apache/streampark/common/fs/HdfsOperatorTest.scala]: There is currently no e2e test, so you can test whether your results are correct.

**Comment by caicancai** [comment]
@zzzk1 Thanks for your contribution. Maybe we can add tests one by one
Because there is no e2e test for similar hdfs, it is certainly not verifiable, but others may be

File: streampark-common/src/test/scala/org/apache/streampark/common/util/DateUtilsTest.scala [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import java.time.{Instant, LocalDateTime}
+import java.util.{Date, TimeZone}
+import java.util.concurrent.TimeUnit
+
+class DateUtilsTest {
+  val dateTestCase = "2000-01-01 00:00:01"
+  val timeStampTestCase: Long = 946656001000L;
+
+  @Test def stringToDateTest(): Unit = {
+    val sdf: SimpleDateFormat = new SimpleDateFormat(DateUtils.fullFormat)
```
Comment (caicancai) [inline streampark-common/src/test/scala/org/apache/streampark/common/util/DateUtilsTest.scala]: You can try other forms of testing, For example, https://github.com/apache/incubator-streampark/tree/dev/streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/test/scala/org/apache/streampark/flink/kubernetes/v2

**Review by caicancai — CHANGES_REQUESTED** [review]

**Comment by zhengkezhou1** [comment]
Test cases may not take into account time zones

**Comment by caicancai** [comment]
> Test cases may not take into account time zones

You can leave comments in your code

**Comment by caicancai** [comment]
Good job, i will review it later

**Pull request closed without merging**